src/main/java/org/inferred/freebuilder/processor/excerpt/CheckedBiMap.java

```diff
@@ -126,9 +128,11 @@ public void addTo(SourceBuilder code) {
           .addLine("  }")
           .addLine("")
           .addLine("  @Override public V setValue(V value) {")
+          .addLine("    final K oldKey = biMap.inverse().get(value);")
           .addLine("    %s.checkArgument(", Preconditions.class)
-          .addLine("        !biMap.containsValue(value), \"value already present: %%s\", value);")
-          .addLine("    V oldValue = value;")
+          .addLine("        oldKey == null || %s.equals(oldKey, key), \"value already present: %%s\", value);",
+              Objects.class)
+          .addLine("    V oldValue = this.value;")
```

Comment:
note: this is separate from the original issue, but I noticed oldValue was using the passed in value and not the previous one.

Reply:
Could you add a test please?

Reply:
Yup added tests for put and setValue and their respective inverse.

Reply:
Still need a test that the value returned from setValue is the old value not the new one